### PR TITLE
ENH: Add pytest and `babs-check-setup`; enhance `babs-init`

### PR DESCRIPTION
**Pull request by zhao-cy**

### CODE REVIEW

File: docs/source/babs-init.rst [hunk 1]
```diff
@@ -1,9 +1,53 @@
...
 
 .. argparse::
    :ref: babs.cli.babs_init_cli
    :prog: babs-init
-   :nodefault:
-   :nodefaultconst:
\ No newline at end of file
+
+    --input : @after
+      Notes to add here: --------
```
Comment (tsalo) [inline docs/source/babs-init.rst:16]:
```suggestion
    --input : @after
      Notes to add here.
```

Comment (tsalo) [reply docs/source/babs-init.rst:16]: Just wondering if the colon and dashes are causing problems.

File: docs/source/babs-init.rst [hunk 2]
```diff
@@ -1,9 +1,53 @@
...
+**********************
 
 .. argparse::
    :ref: babs.cli.babs_init_cli
    :prog: babs-init
-   :nodefault:
-   :nodefaultconst:
\ No newline at end of file
+
+    --input : @after
```
Comment (tsalo) [inline docs/source/babs-init.rst:15]:
```suggestion
   --input : @after
```

It turns out the problem was the indentation.

Comment (zhao-cy) [reply docs/source/babs-init.rst]: Aha! You're right! Thank you so much Taylor!!! That's really hard to catch!!!

**Comment by djarecka** [comment]
@zhao-cy - I've noticed that you changed the way you're providing the container and my previous command doesn't work. Can you please tell me what is your new idea?
I think it would be also nice if the error is provided a bit sooner, when checking the argument provided by the user. Right now I'm getting an error from `sanity_check` that tries to find the container in the directory that the scripts creates, i.e. in `analysis/containers/.datalad/environments/not_really_an_image.simg/image`, so it's hard to figure out where should I put it the container.

**Comment by djarecka** [comment]
fyi. one of the test does not work on my laptop (osx): `test_babs_init[toybidsapp-BIDS-single-ses-True-False]`

```
        if incomplete_results:
>           raise IncompleteResultsError(
                failed=incomplete_results,
                msg="Command did not complete successfully")
E           datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 1 failed:
E           [{'action': 'install',
E             'message': ('Failed to clone from any candidate source URL. Encountered '
E                         'errors per each url were:\n'
E                         '- %s',
E                         'osf://t8urc\n'
E                         "  CommandError: 'git -c diff.ignoreSubmodules=none clone "
E                         '--progress osf://t8urc '
E                         "/private/var/folders/vs/0_30yzfn05d4bj0zpbs4lv140000gn/T/pytest-of-dorota/pytest-6/BIDS0' "
E                         "failed with exitcode 128 [err: 'Cloning into "
E                         "'/private/var/folders/vs/0_30yzfn05d4bj0zpbs4lv140000gn/T/pytest-of-dorota/pytest-6/BIDS0'...\n"
E                         "git: 'remote-osf' is not a git command. See 'git --help'.']"),
E             'path': '/private/var/folders/vs/0_30yzfn05d4bj0zpbs4lv140000gn/T/pytest-of-dorota/pytest-6/BIDS0',
E             'source': {'default_destpath': 't8urc',
E                        'giturl': 'osf://t8urc',
E                        'source': 'osf://t8urc',
E                        'type': 'giturl',
E                        'version': None},
E             'source_url': 'https://osf.io/t8urc/',
E             'status': 'error',
E             'type': 'dataset'}]

../../miniconda3/envs/babs_dev/lib/python3.9/site-packages/datalad/interface/base.py:940: IncompleteResultsError
-
```

Can try to figure out more next week.

**Comment by zhao-cy** [comment]
In this PR, I've changed these things:

* add `babs-check-setup` and add its CLI documentation
* enhance `babs-init` so that it will fail if there is existing BABS project; if it fails it will remove the BABS project it created
* initialize `pytest` (not finished yet for `babs-init`);
* use an updated docker image for CircleCI tests: `pennlinc/babs_tests`
* add more documentations, including how to create a BABS project etc

**Comment by zhao-cy** [comment]
Hi @djarecka ! Regarding your second error about pytest failure, I'm thinking if you've got up-to-date `datalad-osf` or so on your machine. Could you please let me know versions of your datalad, git, git-annex, and datalad-osf? In addition, could you please try running `datalad clone https://osf.io/t8urc/` to see if you'll get any error messages?

Thanks,
Chenying

**Pull request merged**